### PR TITLE
[LY-3176] Add the code for generating the parsec-client

### PR DESCRIPTION
**Pull request by yamlin**

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: parsec-gradle-plugin/build.gradle [hunk 1]
```diff
@@ -51,16 +52,28 @@ postProcess << {
...
 
-task downloadParsecRdl(type: Exec){
-    commandLine "sh", "./download_parsec_rdl.sh", project.property("parsecRdlVersion"), projectDir.absolutePath
+task downloadParsecRdl(type: Exec) {
+//    commandLine "sh", "./download_parsec_rdl.sh", project.property("parsecRdlVersion"), projectDir.absolutePath
+//    outputs.upToDateWhen {
+//        //Return upToDate when parsec_rdl version matches
+//        def line = new File("${projectDir.absolutePath}/src/main/resources/rdl-bin/metadata.xml").readLines().get(0)
+//        return (line && line == project.property("parsecRdlVersion"))
+//    }
```
Comment (joejoe321321) [inline parsec-gradle-plugin/build.gradle]: they can be removed

File: parsec-gradle-plugin/build.gradle [hunk 2]
```diff
@@ -51,16 +52,28 @@ postProcess << {
...
     outputs.upToDateWhen {
-        //Return upToDate when parsec_rdl version matches
-        def line = new File("${projectDir.absolutePath}/src/main/resources/rdl-bin/metadata.xml").readLines().get(0)
-        return (line && line == project.property("parsecRdlVersion"))
+        //Return upToDate when rdl_gen version matches
+        def flag = false
+        try {
+            def line = new File("${projectDir.absolutePath}/src/main/resources/rdl-gen/metadata.xml").readLines().get(0)
+            flag = (line && line == project.property("parsecRdlGeneratorVersion"))
+        } finally {
```
Comment (chinkong83) [inline parsec-gradle-plugin/build.gradle:69]: Why use finally?

Comment (yamlin) [reply parsec-gradle-plugin/build.gradle:69]: I just want to check the version number. When something goes error, I will directly download the rdl-gen.

**Pull request merged**